helpers/helpers.go

```diff
@@ -153,6 +156,70 @@ func GetIncludedBundles(filename string) ([]string, error) {
 	return includes, nil
 }
 
+// UnpackFile unpacks a .tar or .tar.gz/.tgz file to a given directory
+// Should be roughly equivalent to "tar -x[z]f file -C dest"
+func UnpackFile(file string, dest string) error {
+	fr, err := os.Open(file)
+	defer fr.Close()
+	if err != nil {
+		return err
+	}
+
+	var tr *tar.Reader
+
+	// If it's a compressed tarball
+	if strings.HasSuffix(file, ".tar.gz") || strings.HasSuffix(file, ".tgz") {
```

Comment:
FYI: somewhere in swupd we have code that can identify from the file itself whether it is gzipped or not, might make this function more robust. I wouldn't worry about this now, but might be useful to pull it later.

Reply:
Really? That'd be pretty cool. I googled around for a while for a good way to do this in go and eventually gave up.

Reply:
swupd/packs.go

I would defer it for later, as this doesn't seem the main goal of the PR.

Reply:
Just checked it out -- nice! Now my brain is off thinking about how to organize helpers like this that would be useful across all of mixer...

Reply:
> Just checked it out -- nice! Now my brain is off thinking about how to organize helpers like this that would be useful across all of mixer...

If multiple packages are using it the functionality, we either make one of them export, or if it is out of scope for that package, export in a separate package. Depending on the nature of the feature, it might make sense to make it an internal package, i.e. only our packages can import it, but external code not. This separates what we want to reuse internally vs. what we want to export as API to external users. See https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU

(Note: our helpers likely should be an internal package)

YMMV, but is better to have packages with purpose than a grab bag of utilities. (Thinking now, internal packages might reduce the issue with the grab bag style of packages, but unsure...). My strategy is to keep utilities that don't deserve a package for them inside the package that use them. See also the section on bad names https://blog.golang.org/package-names

The concrete answer for this case: we have various archive+compress related code spawning here and there, so soon we might have enough use cases to create a mixer-tools/internal/archive package or similar.

Reply:
It would be nice to not duplicate archive compress/extract in multiple parts if we can re-use and extend 1 package instead.